lib/bundler/shared_helpers.rb

```diff
@@ -152,6 +152,24 @@ def trap(signal, override = false, &block)
       end
     end
 
+    def ensure_same_dependencies(spec, old, new)
+      new = new.reject {|d| d.type == :development }
```

Comment:
this is very nickpicky but can we not call it `new`? It just makes me think that you're initialising a new object